mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -438,7 +446,10 @@ def handle_settings_change(self, message):
             if remote_settings is not None:
                 LOG.info('Updating settings for skill ' + self.name)
                 self.settings.update(**remote_settings)
-                save_settings(self.settings_write_path, self.settings)
+                if isinstance(self.settings, JsonStorage):
+                    self.settings.store()  # else not initialized and it's a dict
+                else:
+                    save_settings(self.settings_path, self.settings)
```

Comment:
maybe these else: clauses should be removed, should be fine to fully deprecate `save_settings`, this will never happen